test/extensions/clusters/dynamic_forward_proxy/cluster_test.cc

```diff
@@ -173,7 +173,7 @@ TEST_F(ClusterTest, InvalidLbContext) {
 }
 
 // Verify that using 'sni' causes a failure.
-TEST_F(ClusterTest, InvalidSNI) {
+TEST_F(ClusterTest, DEPRECATED_FEATURE_TEST(InvalidSNI)) {
```

Comment:
What is going to happen to these tests in `v3`? Can we update them now?

Reply:
These tests are best effort anyway. (i.e. if a TLS transport socket is set inside TAP transport socket the cluster config won't fail) And the override takes priority anyway even SNI is set in UpstreamTlsContext.

Reply:
I'm still a bit confused here. I think either we are OK with stripping these tests completely at the next major version (with resulting potential coverage loss), or we should fix them up while doing the deprecation. It's not like they are doing something that relies inherently on the deprecated TLS context, I think.

Reply:
IMO it's OK to strip these tests at the next major version and the config validation check as well.